pkg/codegen/internal/test/program_driver.go

```diff
@@ -104,62 +173,41 @@ type langConfig struct {
 //nolint: revive
 func TestProgramCodegen(
 	t *testing.T,
-	language string,
-	genProgram func(program *hcl2.Program) (map[string][]byte, hcl.Diagnostics, error),
+	// language string,
+	// genProgram func(program *hcl2.Program) (map[string][]byte, hcl.Diagnostics, error
+	testcase ProgramLangConfig,
+
 ) {
 	for _, tt := range programTests {
 		t.Run(tt.Description, func(t *testing.T) {
-			if tt.Skip.Has(language) {
+			var err error
+			if tt.Skip.Has(testcase.Language) {
 				t.Skip()
 				return
 			}
 
-			expectNYIDiags := false
-			if tt.ExpectNYIDiags.Has(language) {
-				expectNYIDiags = true
-			}
-
-			var cfg langConfig
+			expectNYIDiags := tt.ExpectNYIDiags.Has(testcase.Language)
 
-			switch language {
-			case "python":
-				cfg = langConfig{
-					extension:  "py",
-					outputFile: "__main__.py",
-				}
-			case "nodejs":
-				cfg = langConfig{
-					extension:  "ts",
-					outputFile: "index.ts",
-				}
-			case "go":
-				cfg = langConfig{
-					extension:  "go",
-					outputFile: "main.go",
-				}
-			case "dotnet":
-				cfg = langConfig{
-					extension:  "cs",
-					outputFile: "MyStack.cs",
-				}
-			default:
-				t.Fatalf("language %s not recognized", language)
+			testDir := filepath.Join(testdataPath, tt.Name+"-pp")
```

Comment:
Thank you for restructuring the test cases into directories 👍

Reply:
Might be worth going even further and putting each language's baseline in its own directory ala SDK tests